akka multi threading error

hello, I think this is a very useful tools, thank you very much!
but, I have a question, when I try to add it to akka multi threading, there is a error.
I think "lemmatize"(StanfordLemmatizer.java:30) maybe also need to be singleton, so I change it as singleton, it works.(Morphology.stemStatic => Morphology.stemStaticSynchronized)

error:
[ERROR] [04/10/2016 10:30:52.167] [System-akka.actor.default-dispatcher-15] [akka.actor.LocalActorRefProvider(akka://System)] guardian failed, shutting down system
java.lang.Error: Error: could not match input
	at edu.stanford.nlp.process.Morpha.zzScanError(Morpha.java)
	at edu.stanford.nlp.process.Morpha.next(Morpha.java)
	at edu.stanford.nlp.process.Morphology.lemmatize(Morphology.java:155)
	at edu.stanford.nlp.process.Morphology.stemStatic(Morphology.java:182)
	at it.uniroma1.lcl.adw.utils.StanfordLemmatizer.lemmatize(StanfordLemmatizer.java:30)
	at it.uniroma1.lcl.adw.utils.StanfordLemmatizer.lemmatize(StanfordLemmatizer.java:36)
	at it.uniroma1.lcl.adw.utils.SentenceProcessor.processSentence(SentenceProcessor.java:40)
	at it.uniroma1.lcl.adw.textual.similarity.TextualSimilarity.getStanfordSentence(TextualSimilarity.java:128)
	at it.uniroma1.lcl.adw.textual.similarity.TextualSimilarity.cookSentence(TextualSimilarity.java:266)
	at it.uniroma1.lcl.adw.textual.similarity.PairSimilarity.cookLexicalItem(PairSimilarity.java:124)
	at it.uniroma1.lcl.adw.textual.similarity.PairSimilarity.getSimilarity(PairSimilarity.java:308)
	at it.uniroma1.lcl.adw.ADW.getPairSimilarity(ADW.java:120)
	at main.scala.org.qa.SemanticSimilarityAdwWorker$$anonfun$receive$1.applyOrElse(SemanticSimilarityAdw.scala:33)
	at akka.actor.Actor$class.aroundReceive(Actor.scala:480)
	at main.scala.org.qa.SemanticSimilarityAdwWorker.aroundReceive(SemanticSimilarityAdw.scala:26)
	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:526)
	at akka.actor.ActorCell.invoke(ActorCell.scala:495)
	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:257)
	at akka.dispatch.Mailbox.run(Mailbox.scala:224)
	at akka.dispatch.Mailbox.exec(Mailbox.scala:234)
	at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
	at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
	at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)

thank you!

--- a/src/main/java/it/uniroma1/lcl/adw/utils/StanfordLemmatizer.java
+++ b/src/main/java/it/uniroma1/lcl/adw/utils/StanfordLemmatizer.java
@@ -27,7 +27,7 @@ public class StanfordLemmatizer
 	
 	public WordLemmaTag lemmatize(String word, String tag)
 	{ 
-	    final String lemma = Morphology.stemStatic(word, tag).word();
+	    final String lemma = Morphology.stemStaticSynchronized(word, tag).word();
 	    return new WordLemmaTag(word, lemma, tag);
 	}
 

